docs/DeveloperGuide.md

```diff
@@ -346,74 +280,206 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
+`* * *` | user | add a task with deadlines | record tasks that need to be done
+`* * *` | user | add a floating task | record tasks without specific deadlines
+`* * *` | user |add an event | record tasks that have start time and end time
+`* * *` | user | add a recurring task | record task that will recur at certain frequency
+`* * *` | user | list all task filtered by task types/completion/date/tags| view a specific section of the tasks that have to completed
+`* * *` | user | edit my task | ensure that the task is still relevant
+`* * *` | user | delay my task |automatically postpone deadlines 
+`* * *` | user | delete my task | conveniently remove task that is no longer relevant
+`* * *` | user  | delete all my task | clear my data
+`* * *` | user | delete my task | conveniently remove task that is no longer relevant
+`* * *` | new user | have some flexibility in inputting my command | conveniently remove task that is no longer relevant
+`* * *` | advanced user | utilize text search | find specific tasks with specificied tags or keywords
+`* * *` | user | load my data from any specified location | have ease of access in using the application
+`* * *` | user | export my data to any specified location | share my task list with other local machines
+`* * *` | user | set a deadline reminder | been notified at some time before deadline
+`* * *` | new user | view command lists or more information about a particular command | learn how to use various command
+`* * ` | user | add several tags for a task | search tasks based on tags
+`* * ` | user | sort listed task based on deadline | prioritize the completion of task
+`* * ` | user | find recently deleted tasks | retrieve accidentally deleted tasks and monitor past activities
+`* * ` | user | support for undo/redo operation | flexibly manage the changes made on the list of task
+`*` | user | upload todo items to Google Calendar | synchronize with Google Calendar
+`*` | user | customize application’s background | personalize the appearance of the task manager
+`*` | user | view different task types in different areas on GUI | view different task types easier
+`*` | advanced user | type commands with auto-complete | type a command faster
+`*` | advanced user | alias a particular command with some arguments | type a command faster
 
 {More to be added}
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `BucketList` and the **Actor** is the `user`, unless specified otherwise)
 
-#### Use case: Delete person
+#### Use case: 
 
-**MSS**
+**UC01 ‐ Create a task**
```

Comment:
Good to see you have use cases.
Remember you can follow the interaction in use cases to implement the feature (as actually this how use cases are supposed to work).